### PR TITLE
pkg/umorse: Add Kconfig support

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

This models the `umorse` package in Kconfig.

### Testing procedure

```
make clean -C tests/pkg_umorse/ 
TEST_KCONFIG=1 make menuconfig -C tests/pkg_umorse/ 
TEST_KCONFIG=1 make all test -C tests/pkg_umorse/
```

And check the CI

### Issues/PRs references
Depends on #15984

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
hmmm fails when I try to test with `TEST_KCONFIG=1` I will investigate tomorrow.

**Comment by MrKevinWeiss** [comment]
It seems that it fails due to `xtimer_on_ztimer` not being included.  This mean should probably be fixed either in `posix_sleep` or better yet in the timer modules.

**Comment by MrKevinWeiss** [comment]
So there was some silly dependency issues and compatibility issues with xtimer.  Ultimately the test didn't need xtimer at all since it is using posix_sleep in the package and that uses ztimer now I guess.  So I adapted the test to use the same sleeper as in the package and now the `info-modules` match.

**Comment by MrKevinWeiss** [comment]
I think this should be ready I think, was it causing an issue with the CI @kaspar030 ?

**Comment by MrKevinWeiss** [comment]
hmmm I guess it wasn't ready

**Comment by MrKevinWeiss** [comment]
Added some app.config.test and .murdock tests!

**Comment by MrKevinWeiss** [comment]
I guess the boards don't support the posix sleep header properly, falling back to ztimer as it is still needed vs xtimer.

**Comment by MrKevinWeiss** [comment]
static check fixed, rebased and squashed!

File: pkg/Kconfig [hunk 1]
```diff
@@ -10,5 +10,6 @@ rsource "driver_bme680/Kconfig"
 rsource "semtech-loramac/Kconfig"
 rsource "tinydtls/Kconfig"
 rsource "wakaama/Kconfig"
+rsource "umorse/Kconfig"
```
Comment (leandrolanzieri) [inline pkg/Kconfig:13]: Could we keep alphabetical order?

File: pkg/umorse/Kconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+config PACKAGE_UMORSE
+    bool "Morse code encoder"
```
Comment (leandrolanzieri) [inline pkg/umorse/Kconfig:8]: Maybe we should mention the package name in the prompt? Something like:
```suggestion
    bool "uMorse - Morse code encoder"
```

**Review by leandrolanzieri — COMMENTED** [review]
Just minor comments, otherwise looks good! Feel free to squash directly.

**Comment by leandrolanzieri** [comment]
#16037 has been merged, there is a conflict in the murdock file now :/

**Comment by MrKevinWeiss** [comment]
I will rebase shortly

**Comment by MrKevinWeiss** [comment]
I think she is finally ready

**Review by leandrolanzieri — APPROVED** [review]
ACK!

**Comment by MrKevinWeiss** [comment]
Many thanks for the review @leandrolanzieri

**Pull request merged**